packages/@css-blocks/glimmer/src/ElementAnalyzer.ts

```diff
@@ -167,7 +202,9 @@ export class ElementAnalyzer {
 
     // Find the class or scope attribute and process it
     for (let [namespace, attrName, attribute] of this.eachAnalyzedAttribute(node)) {
-      if (attrName === "class") {
+      // should never be a path expression and "class" as that will throw an error before
+      // guard here is a) to get TS to agree b) just in case something, somehow, slips through.
+      if (attrName === "class" && !isPathExpression(attribute)) {
         this.processClass(namespace, attribute, element, forRewrite);
```

Comment:
A refactor of the code may invalidate the expectations of the comments here. If there's no good way to get the types to match our expectations, we should write this code as if we don't know about checking that is done elsewhere and do it again here...

```suggestion
      if (attrName === "class") {
         if (isPathExpression(attribute)) {
           throw this._missingClassAttributeValue(node, attribute, attrName);
         } else {
           this.processClass(namespace, attribute, element, forRewrite);
         }
```

But that assumes my assertion that "there's no good way to get the types to match our expectations" is true. Is it? I explored that question a bit in this follow-on PR: https://github.com/linkedin/css-blocks/pull/406

Reply:
Yeah, I have no idea if there is a way to get the types to agree here. In theory any `PathExpression` with class should have been removed in the previously called function, but I'm not sure how to convince typescript of that (it _can_ return a `PathExpression` but not one where the `attrName` is `class`)

It's almost like I need to create two types of PathExpression type just so I can mark it as returning the non-class-containing type to convince TS... but that seems a bit... weird.

Reply:
Is there some kind of Union Type magic that can work here? (if input is foo then output string bar will never be x?)